index.html

```diff
@@ -526,55 +526,79 @@
+            Let <var>topLevelOrigin</var> be the URL of the top level page. 
+            Let <var>cachedSupportedMethods</var> be <a>supportedMethods</a> sequences from each <a>PaymentMethodData</a> from previous call to <a>PaymentRequest</a>.
+            Let <var>cachedResponse</var> be response from previous call to <a>canMakePayment</a>.
+            Let <var>canMakePaymentQuotaReached</var> be boolean value of previous call to <a>canMakePayment</a> for the <a>topLevelOrigin</a>.
           </li>
           <li>
-            Let <var>cacheInvalidateTimer</var> be the time <a>user agent</a> recorded for previous call to <a>canMakeActivePayment</a> for <var>topLevelDomain</var>. Set <a>DateTime</a> to <var>cacheInvalidateTimer</var> if it's not set. 
+            Let <var>supportedMethods</var> be the union of all the <a>supportedMethods</a> sequences from each
+            <a>PaymentMethodData</a> in the <var>request</var>@[[\methodData]] sequence.
           </li>
           <li>
-            If <var>cacheInvalidateTimer</var> is set and is still active, resolve <var>acceptPromise</var> promise with <var>cached</var> response.
+            If <var>cachedSupportedMethods</var> is preset and matches <a>supportMethods</a>, 
+            then resolve <var>canMakePaymentPromise</var> with <a>cachedResponse</a>. Else, 
+            set <var>cachedSupportedMethods</var> to <var>supportedMethods</var>.
           </li>
           <li>
-            Let <var>supportedMethods</var> be the union of all the <a>supportedMethods</a> sequences from each
-            <a>PaymentMethodData</a> in the <var>request</var>@[[\methodData]] sequence.
+            If <a>canMakePaymentQuotaReached</a> is set, then reject the <var>canMakePaymentPromise</var>
+            with QuotaExceededError.
           </li>
           <li>
             Let <var>acceptedMethods</var> be <var>supportedMethods</var> with all identifiers removed that the
-            <a>user agent</a> does not accept and method supports active payment instrument for payment.
+            <a>user agent</a> does not support.
           </li>
           <li>
-            If the length of <var>acceptedMethods</var> is zero, then resolve <var>acceptPromise</var> with
+            If the length of <var>acceptedMethods</var> is zero, then resolve <var>canMakePaymentPromise</var> with
             <code>false</code>, otherwise resolve <var>acceptPromise</var> with <code>true</code>.
           </li>
           <li>
-            Cache the response in <var>cachedResponse</var> and set <var>cacheInvalidateTimer</var> to certain <a>DateTime</a> for the <var>topLevelDomain</var>.
+            Cache the response in <var>cachedResponse</var> and set <var>canMakePaymentQuotaReached</var> to 
+            to true.
+          </li>
+          <li> 
+            In addition, implementations may choose to implement a timeout to reset 
+            <var>canMakePaymentQuotaReached</var> for the <var>topLevelOrigin</var>.
```

Comment:
This doesn't really work without a quota reset at some point. I think what we need to specify here is that implementations {should or must} reset the quota after a timeout, with the timeout selected by the implementation.